### ISSUE TITLE
Document MPI/OpenMP/nodes distribution done in `cice.batch.csh`

### ISSUE THREAD

**Issue by phil-blain** [CICE-Consortium/CICE#650]
I often have to re-read the computations at the top of `cice.batch.csh` and do a numeric example to understand what exactly is being done here:

https://github.com/CICE-Consortium/CICE/blob/4147db49dbc93ba93cbb5560b1c299f96a98f4d7/configuration/scripts/cice.batch.csh#L14-L26

It would be nice if those computations could be summarized (and maybe written in a more mathematical form) somewhere in our documentation.

Tangentially, these computation are copy-pasted in `cice.launch.csh`, so we should create a third script (`cice.distrib.csh` ?) and source it from `cice.batch.csh` and `cice.launch.csh` to reduce code duplication.

**Comment by apcraig** [comment]
You're right about this.  My only concern in creating a shared script is that these computations tend to be pretty machine dependent, so while there are some basic things that are shared, there are other machine dependent parts that are not easily shared.  If we unify too much, we changes to get one machine working breaking other machines.  But we should unify where we can.  I'll take a look.

**Comment by phil-blain** [comment]
Well, the computations are exactly the same in cice.launch.csh and cice.batch.csh, that's why I was suggesting unifying these in a new script :)

**Comment by phil-blain** [comment]
Hi Tony, I'm not sure we should have closed this one as one of my points was to document in the user guide what computations were being done by that script.

**Comment by apcraig** [comment]
I have mixed feelings about doing a bunch of documentation.  In part, these calculations are dependent on the syntax needed for each machine.  There are a lot of formats and a number of different "parameters" are used.  As new machines are added, new parameters may be needed.   The syntax for batch and launch are often quite different, which is largely why I thought the calculations should be local to each script.  But there is also a lot of overlap.  I also understand it's a little bit of a pain to decipher the calculations everytime a port is added/updated, but I also think that's just part of the process of porting.  If we want to increase the documentation, I think it should be done as comments where the calculations are done, and I tried to do a bit of that as part of this update.
